samples/performance/constant_data/constant_data_tutorial.md

```diff
@@ -0,0 +1,403 @@
+
+***Note:** To check how much data we can store in uniform buffers and storage buffers, you can query the physical device for its `VkPhysicalDeviceLimits` and check the values `maxUniformBufferRange` and `maxStorageBufferRange` respectively.*
+
+#### *Interface Blocks*
+
+To implement our constant data we have to use an interface block. Interface blocks in shader code are used to group multiple global variables of the same `<storage>` type, so in theory they aren't necessarily solely for constant data.
+
+For example:
+
+```
+layout(set = 0, binding = 0) uniform PerMeshData
+{
+    vec4 camera_position;
+    mat4 model_matrix;
+    vec3 mesh_color;
+}
+per_mesh_data;
+```
+
+Interface blocks are still global variables, and technically still follow the global variable format that was mentioned at the start of this chapter. However, the difference is that they have to be given a user-defined type. They work exactly the same way as a `struct` in GLSL/C++. For example, to access the model matrix in this interface block, you'd use `per_mesh_data.model_matrix`. 
+
+*You can read more about interface blocks [here](https://www.khronos.org/opengl/wiki/Interface_Block_(GLSL)).*
+
+### **Vulkan API**
+
+We've covered how constant data is implemented in the shader, however to push the data from the application to the shader we need to use Vulkan.
+
+We do this with the use of `VkBuffer`s, which is Vulkan's implementation of buffer memory.
+
+Buffers in Vulkan are just chunks of memory used for storing data, which can be read by the GPU.
+
+They need to be created and have their memory *manually* allocated, and then we can copy our constant data into the allocated memory. This data can then be plugged into the draw calls, so that it can finally be used in our shader computations. 
+
+**Note:** *The library [Vulkan Memory Allocator (VMA)](https://github.com/GPUOpen-LibrariesAndSDKs/VulkanMemoryAllocator) is extremely good for handling a lot of the common pitfalls that come with managing your Vulkan memory, without removing the control that you would otherwise have with native Vulkan.*
+
+The following links are useful for learning how to create a Vulkan buffer in your application:
+
+* https://vulkan-tutorial.com/Vertex_buffers/Vertex_buffer_creation
+* https://vulkan-tutorial.com/Vertex_buffers/Staging_buffer#page_Abstracting-buffer-creation
+* https://vulkan-tutorial.com/Uniform_buffers/Descriptor_layout_and_buffer#page_Uniform-buffer
+
+#### *The Methods*
+
+Generally, there are two ways to push your constant data. It is either with **push constants** or **descriptor sets**. 
```

Comment:
Specialization Constants, Inline uniform buffer objects and push descriptors?

I would remove "generally" and just say "there are various ways, but only covering a subset"